features/FEATURE_LWIP/lwip-interface/lwip_stack.c

```diff
@@ -420,44 +472,14 @@ int lwip_bringup(bool dhcp, const char *ip, const char *netmask, const char *gw)
         }
     }
 
-#if LWIP_IPV4
-    if (!dhcp) {
-        ip4_addr_t ip_addr;
-        ip4_addr_t netmask_addr;
-        ip4_addr_t gw_addr;
-
-        if (!inet_aton(ip, &ip_addr) ||
-            !inet_aton(netmask, &netmask_addr) ||
-            !inet_aton(gw, &gw_addr)) {
-            return NSAPI_ERROR_PARAMETER;
-        }
-
-        netif_set_addr(&lwip_netif, &ip_addr, &netmask_addr, &gw_addr);
-    }
-#endif
-
     netif_set_up(&lwip_netif);
 
-#if LWIP_IPV4
-    // Connect to the network
+#if !DEVICE_EMAC
     lwip_dhcp = dhcp;
-
-    if (lwip_dhcp) {
-        err_t err = dhcp_start(&lwip_netif);
-        if (err) {
-            return NSAPI_ERROR_DHCP_FAILURE;
-        }
-    }
-#endif
-
-    // If doesn't have address
-    if (!lwip_get_ip_addr(true, &lwip_netif)) {
-        //ret = sys_arch_sem_wait(&lwip_netif_has_addr, 15000);
-        ret = sys_arch_sem_wait(&lwip_netif_has_addr, 30000);
-        if (ret == SYS_ARCH_TIMEOUT) {
-            return NSAPI_ERROR_DHCP_FAILURE;
-        }
-        lwip_connected = true;
+    if (!dhcp) {
+        lwip_start_static_ip(ip, netmask, gw);
+    } else {
+        lwip_start_dhcp(DHCP_TIMEOUT);
```

Comment:
Ok, but it can't stay where it was as WiFi won't have address at this stage. I'm not sure what's the right solution will look into it.